File: test-suite/src/test/java/io/micronaut/microstream/docs/CustomerSave.java
package io.micronaut.microstream.docs;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

import jakarta.validation.constraints.NotBlank;

@Serdeable
public class CustomerSave {

    @NonNull
    @NotBlank
    private String firstName;

    @Nullable
    private String lastName;

    public CustomerSave(@NonNull String firstName, @Nullable String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    @NonNull
    public String getFirstName() {
        return firstName;
    }

    @Nullable
    public String getLastName() {
        return lastName;
    }
}

File: test-suite-python/src/test/python/micronaut/microstream/docs/CustomerSave.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank
from micronaut.serde.annotation import Serdeable


@Serdeable
@dataclass
class CustomerSave:
    firstName: Annotated[str, NotBlank]
    lastName: str | None = None
